addons/mail/static/src/core/web/activity_service.js

```diff
@@ -52,29 +52,46 @@ export class ActivityService {
         return action;
     }
 
-    async schedule(resModel, resId, activityId = false, defaultActivityTypeId = undefined) {
+    async edit(activityId){
+        return new Promise((resolve) =>
+            this.env.services.action.doAction(
+                {
+                    type: "ir.actions.act_window",
+                    name: _t("Schedule Activity"),
+                    res_model: "mail.activity",
+                    view_mode: "form",
+                    views: [[false, "form"]],
+                    target: "new",
+                    res_id: activityId,
+                },
+                { onClose: resolve }
+            )
+        );
+    }
+
+    async schedule(resModel, resIds, defaultActivityTypeId = undefined) {
         const context = {
             default_res_model: resModel,
-            default_res_id: resId,
+            default_res_ids: resIds.map((resId) => `${resId}`).join(','),
```

Comment:
As proposed above, would make a list, so that we always handle resIds as being a list (in python) or an array (in JS)... easier for me. You can write a list directly into a char field in py (we do it in the various composers models).